stacker/blueprints/base.py

```diff
@@ -152,6 +173,95 @@ def setup_parameters(self):
             p = build_parameter(param, attrs)
             t.add_parameter(p)
 
+    def defined_parameters(self):
+        """Return a dictionary of parameters defined by the blueprint.
+
+        By default, this will just return the values from
+        `BLUEPRINT_PARAMETERS`, but this makes it easy for subclasses to add
+        parameters.
+
+        Returns:
+            dict: parameters defined by the blueprint
+
+        """
+        return getattr(self, "BLUEPRINT_PARAMETERS", {})
+
+    def get_parameters(self):
+        """Return a dictionary of parameters available to the template.
+
+        These parameters will have been defined within `BLUEPRINT_PARAMETERS`
+        or `self.defined_parameters`. Any parameter value that depends on stack
+        output will have been resolved.
+
+        Returns:
+            dict: parameters available to the template
+
+        """
+        if self.resolved_parameters is None:
```

Comment:
I didn't want `get_parameters` to take any arguments and was invoking `resolve_parameters` where we were before: https://github.com/remind101/stacker/pull/194/files#diff-92ed81f5017042692188db7b5961c619R203. We need `stack.parameters`, `provider` and `context` which was readily available there, i'm not sure if we can get all of those values when we init the blueprint to make this resolve when called.